tests/testthat/test-query-join.R

```diff
@@ -20,6 +20,12 @@ test_that("only disambiguates shared variables", {
   expect_snapshot(left_join(lf1, lf2, by = c("y" = "z")))
 })
 
+test_that("disambiguate variables that only differ in case", {
+  lf1 <- lazy_frame(x = 1, y = 2)
+  lf2 <- lazy_frame(X = 1, y = 2)
+  expect_snapshot(left_join(lf1, lf2, by = "y"))
```

Comment:
I think it would be marginally clearer to check the column names here. It's always better to avoid a snapshot test if at all possible.

Reply:
Unfortunately, it is not enough to check the column names

Before the PR

``` r
library(dplyr)
library(dbplyr)

lf1 <- lazy_frame(x = 1, y = 2)
lf2 <- lazy_frame(X = 1, y = 2)

op_vars(left_join(lf1, lf2, by = "y"))
#> [1] "x" "y" "X"
show_query(left_join(lf1, lf2, by = "y"))
#> <SQL>
#> SELECT `x`, `LHS`.`y` AS `y`, `X`
#> FROM `df` AS `LHS`
#> LEFT JOIN `df` AS `RHS`
#> ON (`LHS`.`y` = `RHS`.`y`)
```

<sup>Created on 2021-12-09 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

The column names are already correct, the issue is the `SELECT` clause. After the PR it is

```sql
SELECT `LHS`.`x` AS `x`, `LHS`.`y` AS `y`, `RHS`.`X` AS `X`
```

So, the most direct, alternative test would be something like

```r
lf1 <- lazy_frame(x = 1, y = 2)
lf2 <- lazy_frame(X = 1, y = 2)

con <- simulate_dbi()
expect_equal(
  sql_join_var(
    con,
    alias = "x",
    x = "x",
    y = NA_character_,
    all_x = c("x", "y"),
    all_y = c("X", "y")
  ),
  sql(sql_table_prefix(con, "x", "LHS"))
)
```